tests/integration/devfile/cmd_devfile_push_test.go

```diff
@@ -633,6 +633,7 @@ var _ = Describe("odo devfile push command tests", func() {
 			// 1) Push a generic Java project
 			helper.CmdShouldPass("odo", "create", "java-springboot", "--project", namespace, cmpName)
 			helper.CopyExample(filepath.Join("source", "devfiles", "springboot", "project"), context)
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "springboot", "devfile.yaml"), filepath.Join(context, "devfile.yaml"))
```

Comment:
Why we can't use devfile coming from odo devfile registry?

Reply:
the devfile is recently updated, in this test we are checking for this particular string `helper.WaitForOutputToContain("Non-readable POM", 180, 10, session)`
which is not coming in the new devfile.

Reply:
this command `/artifacts/bin/build-container-full.sh` in the old springboot devfile uses `mvn -Dmaven.repo.local=/data/cache/.m2/repository -f ./pom.xml package -Dmaven.test.skip=true` which was explicitly specifying `pom.xml`, so this is generating the string `Non-readable POM` which is not generated by new devfile. 

IMO the check in the test needs to be updated, but for quick fix we can use the odo repo devfile.
Test i would update with this PR https://github.com/openshift/odo/pull/3799

Reply:
ok